### DISCUSSION TITLE
What is the easiest way to use this repository to spawn self-hosted runners?

### DISCUSSION THREAD

**Discussion by domvo** [actions/virtual-environments#7130] [General questions]
I would like to use the information in this repository to spin up a self hosted github actions runner. I have on-premise machines or aws ec2 instances that I can use for that. 

I see a lot of powershell files that look like they are used to be run in Azure. I see some Hashicorp hcl files - can they be used to build the images with packer?

I'd really appreciate just a few hints on how to make it work. Thanks!

**Comment by nnellans** [comment]
See the instructions:  https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md

There are 2 main scripts provided in this repo:
- The first one is to help you run Packer to build the VHD image.  You must run it on a machine that has packer, powershell, azure cli, etc. (the full requirements are in the instructions).  The problem with this script is that it currently creates a VHD, which will eventually be deprecated, as opposed to a Managed Image or a Compute Gallery Image Version.
- The second script is meant to help you create a new Azure VM using the VHD you built from the first script.

After you have created a VM from the image, it is still up to you to install the Runner software and connect it to your GitHub.

**Comment by pkit** [comment]
@domvo 
You will need to build the Azure vhd by using the supplied [scripts](https://github.com/actions/runner-images/blob/main/docs/create-image-and-azure-resources.md)
Then copy that vhd to S3 (beware, it's **big asf** ~80GB)
Then import it as an AMI with [vm-import](https://docs.aws.amazon.com/vm-import/latest/userguide/vmimport-image-import.html)
Then you can run it on AWS EC2.

To run locally:
- download the VHD from Azure
- convert it to `qcow2` format (it's a `raw` vhd)
- run it in KVM locally
